contingency/contingency-api/src/main/java/com/powsybl/contingency/ContingencyContext.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.contingency;
+
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ * @author Etienne Lesot <etienne.lesot at rte-france.com>
+ *
+ * provide the context to get information of the network after a security analysis
+ * it contains a contingency's id and a context type. Context type defines
+ * if we want the information on N situation, N-k situation or both.
+ * contingency's id is defined if informations are needed after
+ * a specific contingency computation
+ *
+ */
+public class ContingencyContext {
+
+    private final String contingencyId;
+
+    /**
+     * Define if informations needed are in N_Situation, N-k Situation or both
+     * if it is N_situation contingencyid is null
+     */
+    private final ContingencyContextType contextType;
+
+    public ContingencyContext() {
```

Comment:
I would remove that constructor (not explicit enough: what  does it create ?). The `none()` method can be used instead by the user.

Reply:
I need a default constructor for serialization

Reply:
Could you elaborate? Is it for future JSON serialization?

Reply:
There is already some JON serialization used to read from input file in security analysis tool and to write down input files for distributed security analysis.

It's very strange : it seems that jackson is able to change `private final` fields after construction ...
I would propose to either make this constructor private, or say in javadoc that it builds a "NONE" context, or remove the constructor and annotate the other one with jackson annotations (I would prefer that more explicit solution).

Reply:
I agree, I think it is better to remove the constructor and annotate the other one.